### PR TITLE
Move the battery attributes down into libfwupd

### PR DESCRIPTION
**Pull request by hughsie**
This allows us to show the current value and the threshold on the CLI
and in tools like gnome-firmware.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupd/fwupd-device.c [hunk 1]
```diff
@@ -3183,6 +3338,38 @@ fwupd_device_class_init(FwupdDeviceClass *klass)
 				    NULL,
 				    G_PARAM_READWRITE | G_PARAM_STATIC_NAME);
 	g_object_class_install_property(object_class, PROP_UPDATE_IMAGE, pspec);
+
+	/**
+	 * FwupdDevice:battery-level:
+	 *
+	 * The device battery level in percent.
+	 *
+	 * Since: 1.5.8
```
Comment (superm1) [inline libfwupd/fwupd-device.c:3347]: 1.8.1

File: libfwupd/fwupd-device.c [hunk 2]
```diff
@@ -3183,6 +3338,38 @@ fwupd_device_class_init(FwupdDeviceClass *klass)
...
+				  FWUPD_BATTERY_LEVEL_INVALID,
+				  G_PARAM_READWRITE | G_PARAM_STATIC_NAME);
+	g_object_class_install_property(object_class, PROP_BATTERY_LEVEL, pspec);
+
+	/**
+	 * FwupdDevice:battery-threshold:
+	 *
+	 * The device battery threshold in percent.
+	 *
+	 * Since: 1.5.8
```
Comment (superm1) [inline libfwupd/fwupd-device.c:3363]: 1.8.1

File: libfwupd/fwupd-enums-private.h [hunk 3]
```diff
@@ -482,5 +482,21 @@ G_BEGIN_DECLS
...
+ *
+ * Result key to represent the battery level in percent.
+ *
+ * The D-Bus type signature string is 'u' i.e. a unsigned 32 bit integer.
+ **/
+#define FWUPD_RESULT_KEY_BATTERY_LEVEL "BatteryLevel"
+/**
+ * FWUPD_RESULT_KEY_BATTERY_THRESHOLD:
+ *
+ * Result key to represent the battery threshold in percent.
```
Comment (superm1) [inline libfwupd/fwupd-enums-private.h:496]: This should have more description. Threshold of "what"

**Review by superm1 — COMMENTED** [review]
Just a few nits

File: libfwupd/fwupd-enums-private.h (hunk 3 above)
Comment (hughsie) [reply libfwupd/fwupd-enums-private.h:496]: `Result key to represent the battery threshold of the permitted battery level in percent` perhaps?

Comment (superm1) [reply libfwupd/fwupd-enums-private.h:496]: I would say something like
`Result key to represent the minimum battery level required to perform an update.  Expressed from 0-100 %.`

**Review by superm1 — APPROVED** [review]

**Pull request merged**